src/auth/useAuth.ts

```diff
@@ -15,16 +18,31 @@ export const useAuth = () => {
   useEffect(() => {
     let isMounted = true;
     const auth = getAuth();
-    const unsubscribe = onAuthStateChanged(auth, async (user) => {
+    const authStateUnsubscribe = onAuthStateChanged(auth, async (user) => {
       if (!isMounted) return;
       setLoading(true);
       setUser(user);
       const result = user && (await getIdTokenResult(user, true));
       setClaims(result?.claims || null);
       setLoading(false);
     });
+
+    const idTokenUnsubscribe = withCookie
+      ? onIdTokenChanged(auth, async (user) => {
```

Comment:
`?`じゃなくて`&&`にしちゃえば`: undefind`なくせそう

Reply:
それだとfalseになるから、結局idTokenUnsubscribe実行のところで型チェックのif文が増えると思うけどいいやり方ある？

Reply:
あ、確かに。

Reply:
```suggestion
         (withCookie || undefined) &&  onIdTokenChanged(auth, async (user) => {
```

こう書けばいけるけど、これはこれでみづいよね。

Reply:
だね。元のままでよいと思います 🙏